### PR TITLE
[skip ci] export_slow_tests.py - Add option to ignore small differences 

### PR DESCRIPTION
**Pull request by janeyx99**
This would lower the number of unnecessary commits to pytorch/test-infra by only exporting a different stats file when the stats are varying enough. This way, if the slow test cases we gather from S3 are the same and their times are trivially different, then we do not bother exporting a different stats file when the --ignore-small-diffs option is enabled.

We instead export the stats already in test-infra, so that when it tries to commit, it realizes it would be an empty commit and not add to the git history.

Test plan:
Run `python tools/export_slow_tests.py --ignore-small-diffs <threshold>`.

### CODE REVIEW

**Comment by driazati** [comment]
Is there any reason we check this file into git vs keeping it all in S3? That would let us update as often as we want without worrying about polluting the commit history (may also be faster to download over github's servers? idk). It is nice that there's an easy audit trail + we can make manual edits if necessary without extra tooling though

**Comment by janeyx99** [comment]
> Is there any reason we check this file into git vs keeping it all in S3? That would let us update as often as we want without worrying about polluting the commit history (may also be faster to download over github's servers? idk). It is nice that there's an easy audit trail + we can make manual edits if necessary without extra tooling though

Yea we've considered this before--you got it that the main reason is so it's more easily and quickly readable and update-able by hand in case anything is off.

File: tools/export_slow_tests.py [hunk 1]
```diff
@@ -37,11 +38,38 @@ def filter_slow_tests(test_cases_dict: Dict[str, float]) -> Dict[str, float]:
...
+    return cast(Dict[str, float], json.loads(contents))
+
+
+def too_similar(calculated_times: Dict[str, float], other_times: Dict[str, float], threshold: float) -> bool:
+    # check that their keys are the same
+    if calculated_times.keys() != other_times.keys():
+        return False
+
+    for test_case, test_time in calculated_times.items():
+        if test_case not in other_times:
```
Comment (driazati) [inline tools/export_slow_tests.py:53]: isn't this not possible given the `calculated_times.keys() != other_times.keys():` check above?

File: tools/export_slow_tests.py [hunk 2]
```diff
@@ -37,11 +38,38 @@ def filter_slow_tests(test_cases_dict: Dict[str, float]) -> Dict[str, float]:
...
+            print(test_case)
```
Comment (driazati) [inline tools/export_slow_tests.py:54]:
```suggestion
```

**Review by driazati — APPROVED** [review]

File: tools/export_slow_tests.py (hunk 1 above)
Comment (janeyx99) [reply tools/export_slow_tests.py:53]: Ah yes, I forgot to remove debugging code oops!

**Review by walterddr — APPROVED** [review]

**Pull request closed without merging**